Public/Src/Engine/Scheduler/Artifacts/FileContentManager.cs

```diff
@@ -1531,8 +1531,20 @@ public static bool CreateSymlinkEagerly(LoggingContext loggingContext, IConfigur
                 }
             }
 
-            if (IsDistributedWorker)
+            // Delete the absent files if any
+            if (!await DeleteFilesRequiredAbsentAsync(state, operationContext))
             {
+                return ArtifactMaterializationResult.DeleteFilesRequiredAbsentFailed;
+            }
+
+            if (IsDistributedWorker && SourceFileMaterializationEnabled)
+            {
+                // NOTE: In the typical case (SourceFileMaterializationEnabled=false), we only verify source files after
+                // placement to ensure directory symlinks which can make outputs appear as source files work. This relies
+                // on the logic that pips which depend on the source file version of a file under a directory symlink also
+                // have a dependency on the corresponding file at its real location where it appears as an output and therefore
+                // will be materialized during PlaceFilesAsync.

```

Comment:
This need a better comment. Perhaps provide an example to make the scenario easily understandable.